code/modules/unit_tests/explosion_action.dm

```diff
@@ -0,0 +1,214 @@
+
+	// invert the order of ourselves setting up the `/turf/closed` checks because we don't automatically start off as an open turf here
+	var/original_closed_turf_type = run_loc_floor_top_right.type // should just be /turf/closed/wall but lets be hardy against changes to the map template should they arise
+	var/original_closed_baseturfs = islist(run_loc_floor_top_right.baseturfs) ? run_loc_floor_top_right.baseturfs.Copy() : run_loc_floor_top_right.baseturfs
+	var/turf/closed/wall/test_closed_turf = run_loc_floor_top_right.ChangeTurf(CLOSED_FLOOR_TYPE)
+
+	EX_ACT(test_closed_turf, EXPLODE_NONE, test_closed_turf) // regardless of severity, this should dismantle the wall
+	TEST_ASSERT_NOTEQUAL(test_closed_turf.type, CLOSED_FLOOR_TYPE, "EX_ACT() with EXPLODE_NONE severity (setting itself as the target) should have eviscerated the wall, but instead saw zero changes!")
+	test_closed_turf.ChangeTurf(original_closed_turf_type, original_closed_baseturfs)
+
+	test_closed_turf.ChangeTurf(CLOSED_FLOOR_TYPE)
+	var/cached_hardness = test_closed_turf.hardness
+	test_closed_turf.hardness = 100 // we'll set the hardness to 100 so we don't get errant failures
+	EX_ACT(test_closed_turf, EXPLODE_LIGHT)
+	TEST_ASSERT_NOTEQUAL(test_closed_turf.type, CLOSED_FLOOR_TYPE, "EX_ACT() with EXPLODE_LIGHT severity should have dismantled the wall, but instead saw zero changes!")
+	test_closed_turf.ChangeTurf(original_closed_turf_type, original_closed_baseturfs)
+
+	// just to make sure the hardness is what we wanted it to be, there's no real reason why the hardness should inherit between changeturfs but by God we should guard against it.
+	test_closed_turf.ChangeTurf(CLOSED_FLOOR_TYPE)
+	test_closed_turf.hardness = cached_hardness
+	EX_ACT(test_closed_turf, EXPLODE_HEAVY) // wall will be dismantled at the very least
+	TEST_ASSERT_NOTEQUAL(test_closed_turf.type, CLOSED_FLOOR_TYPE, "EX_ACT() with EXPLODE_HEAVY severity should have dismantled the wall, but instead saw zero changes!")
+	test_closed_turf.ChangeTurf(original_closed_turf_type, original_closed_baseturfs)
+
+	test_closed_turf.ChangeTurf(CLOSED_FLOOR_TYPE)
+	EX_ACT(test_closed_turf, EXPLODE_DEVASTATE) // yeah we're definitely not seeing the wall anymore
+	TEST_ASSERT_NOTEQUAL(test_closed_turf.type, CLOSED_FLOOR_TYPE, "EX_ACT() with EXPLODE_DEVASTATE severity should have eviscerated the wall, but instead saw zero changes!")
+	test_closed_turf.ChangeTurf(original_closed_turf_type, original_open_baseturfs)
+
+	// to be super duper sooper safe, clean up both turfs we changed a second time to ensure we aren't yonking downstream tests by invoking them on the actual turfs rather than what we casted
+	run_loc_floor_bottom_left.ChangeTurf(original_open_turf_type, original_open_baseturfs)
+	run_loc_floor_top_right.ChangeTurf(original_closed_turf_type, original_closed_baseturfs)
+
+#undef OPEN_FLOOR_TYPE
+#undef CLOSED_FLOOR_TYPE
+
+/// From the base implementation of `/atom`, [var/max_integrity].
+#define MAX_INTEGRITY_VALUE 500
```

Comment:
Why not `initial(atom_thing.max_integrity)`?

If we were on 515 right now it'd be `/atom::max_integrity`

Reply:
i wasn't sure how picky we should be since atom_integrity is considered a private var? i've got a better idea anyhow